src/datasets/parallel/parallel.py

```diff
@@ -0,0 +1,103 @@
+from multiprocessing import Pool, RLock
+from tqdm.auto import tqdm
+from typing import List
+import contextlib
+import joblib
+
+from ..utils import logging
+
+logger = logging.get_logger(__name__)
+
+
+class ParallelBackendConfig:
+    backend_name = None
+    steps = []
+
+
+def parallel_map(function, iterable, num_proc, types, disable_tqdm, desc, single_map_nested_func):
+    """
+    Apply a function to iterable elements in parallel, where the implementation uses either multiprocessing.Pool or
```

Comment:
Could you mark this function as experimental in the docstring ? This way we can still modify it if we want
```suggestion
    **Experimental.** Apply a function to iterable elements in parallel, where the implementation uses either multiprocessing.Pool or
```

Reply:
Maybe we could decorate these functions with a decorator similar to this one: https://github.com/huggingface/huggingface_hub/blob/0410930a5b5e24fa33da3a0e9d14c96c4f6e2b44/src/huggingface_hub/utils/_experimental.py#L15

Reply:
ah good idea !